### PR TITLE
Add my handwired keyboard 2d4d

### PR DESCRIPTION
**Pull request by mechmehmet**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by waffle87** [comment]
Unsure why you closed #20105 , but the suggestions I made still need to be applied.
* Everything in `config.h` file needs to be migrated to `info.json` file, as per the [PR checklist](https://docs.qmk.fm/#/pr_checklist?id=keyboard-prs). See [info.json reference](https://docs.qmk.fm/#/reference_info_json) as well.
* The layout macro in `2d4d.h` file also needs to be migrated to `info.json`. That file can then be deleted.
* Basic encoder functionality needs to exist in `2d4d.c` file. See [doc](https://docs.qmk.fm/#/feature_encoders?id=callbacks).

You can use `qmk info -f json -kb  handwired/2d4d` to help automate some of this.

**Comment by mechmehmet** [comment]
Thanks for your help, but why didnt you read my comments, I have not a basic encoder func on keymap and if you look at the doc, it says you can use on keymap with return false. Can you check my keymap for encoder func and layer_state_t function; case 6. I cannot use in c file, how could I use.

**Comment by mechmehmet** [comment]
@waffle87 I merge all files to info.json, but you can tell me: "we aimed to use info.json instead of struggle with other files". At the end of 3 hours, I could only understand what was wrong. When I compare the old projects, nothing is understood, because I already did it by taking examples from the projects.

File: keyboards/handwired/2d4d/info.json [hunk 1]
```diff
@@ -0,0 +1,127 @@
+{
+    "keyboard_name": "2d4d",
+    "maintainer": "qmk",
+    "manufacturer": "mechMehmet",
+    "board": "BLACKPILL_STM32_F401",
+    "bootloader": "stm32-dfu",
+    "bootmagic": {
+        "enabled": false
+    },
```
Comment (waffle87) [inline keyboards/handwired/2d4d/info.json:9]:
```suggestion
```

File: keyboards/handwired/2d4d/info.json [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        "encoder": true,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": false,
+        "rgblight": false
+    },
+    "debounce": 5,
+    "diode_direction": "COL2ROW",
+    "encoder": {
+        "enabled": true,
```
Comment (waffle87) [inline keyboards/handwired/2d4d/info.json:30]:
```suggestion
    "features": {
        "encoder": true,
        "extrakey": true,
        "mousekey": true
    },
    "diode_direction": "COL2ROW",
    "encoder": {
```

File: keyboards/handwired/2d4d/info.json [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    "platform": "STM32",
+    "platform_key": "chibios",
+    "processor": "STM32F401",
+    "processor_type": "arm",
+    "protocol": "ChibiOS",
+    "layouts": {
+        "LAYOUT_8x6": {
+            "c_macro": true,
+            "filename": "keyboards/handwired/2d4d/2d4d.h",
+            "json_layout": false,
```
Comment (waffle87) [inline keyboards/handwired/2d4d/info.json:73]:
```suggestion
    "matrix_pins": {
        "cols": ["B12", "B13", "B14", "B15", "A8", "A15"],
        "rows": ["B3", "B4", "B5", "B6", "A2", "A3", "A4", "A5"]
    },
    "processor": "STM32F401",
    "layouts": {
        "LAYOUT_8x6": {
```

File: keyboards/handwired/2d4d/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,455 @@
...
+
+#include "keymap_turkish_f.h"
+#include "sendstring_turkish_f.h"
+
+uint8_t laughVal = 1;
+uint8_t ahahCount = 0;
+uint8_t incTemp = 0;
+enum custom_keycodes {
+    SHUT = SAFE_RANGE, 
+
```
Comment (waffle87) [inline keyboards/handwired/2d4d/keymaps/default/keymap.c:28]:
```suggestion
    SHUT = QK_USER
```

File: keyboards/handwired/2d4d/config.h [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+/* disable debug print */
+//#define NO_DEBUG
+
+/* disable print */
+//#define NO_PRINT
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
```
Comment (waffle87) [inline keyboards/handwired/2d4d/config.h:53]:
```suggestion
/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
#define LOCKING_SUPPORT_ENABLE
/* Locking resynchronize hack */
#define LOCKING_RESYNC_ENABLE

#define ledGreen B0
#define ledBlue  B1
#define ledRed   B10
#define ledWhite B9
```

File: keyboards/handwired/2d4d/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,455 @@
...
+    return state;
+}
+ 
+
+bool led_update_kb(led_t led_state) {
+    if (led_update_user(led_state)) {
+        //writePin(NUM_LOCK_LED_PIN, led_state.num_lock);
+        writePin(ledWhite, !led_state.caps_lock);
+        //writePin(SCROLL_LOCK_LED_PIN, led_state.scroll_lock);
+    }
```
Comment (waffle87) [inline keyboards/handwired/2d4d/keymaps/default/keymap.c:265]:
```suggestion
    writePin(ledWhite, !led_state.caps_lock);
```

File: keyboards/handwired/2d4d/rules.mk [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+MCU = STM32F401
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+
+
```
Comment (waffle87) [inline keyboards/handwired/2d4d/rules.mk:11]:
```suggestion
# This file intentionally left blank
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/handwired/2d4d/info.json [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+                "pin_b": "A0",
+                "resolution": 4
+            }
+        ]
+    },
+    "matrix_pins": {
+        "cols": ["B12", "B13", "B14", "B15", "A8", "A15"],
+        "rows": ["B3", "B4", "B5", "B6", "A2", "A3", "A4", "A5"]
+    },
+    "processor": "STM32F401",
```
Comment (drashna) [inline keyboards/handwired/2d4d/info.json:31]:
```suggestion
```

File: keyboards/handwired/2d4d/info.json [hunk 9]
```diff
@@ -0,0 +1,87 @@
+{
+    "keyboard_name": "2d4d",
+    "maintainer": "qmk",
+    "manufacturer": "mechMehmet",
+    "board": "BLACKPILL_STM32_F401",
+    "bootloader": "stm32-dfu",
```
Comment (drashna) [inline keyboards/handwired/2d4d/info.json:6]:
```suggestion
    "development_board": "blackpill_f401",
```

File: keyboards/handwired/2d4d/keymaps/default/keymap.c
Comment (drashna) [inline keyboards/handwired/2d4d/keymaps/default/keymap.c:1]: In general, we want the default keymap to be simple and clean.  Additional stuff should go into a separate keymap.

File: keyboards/handwired/2d4d/readme.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
+# 2d4d
+
+![2d4d](https://i.imgur.com/famGFG8.jpg)
+
+![2d4d_2](https://i.imgur.com/IfoQUnc.jpg)
```
Comment (drashna) [inline keyboards/handwired/2d4d/readme.md:5]:
```suggestion
![2d4d](https://i.imgur.com/famGFG8h.jpg)

![2d4d_2](https://i.imgur.com/IfoQUnch.jpg)
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/2d4d/keymaps/default/keymap.c
Comment (drashna) [reply keyboards/handwired/2d4d/keymaps/default/keymap.c:1]: `layer_state_set_kb`.

Also, moving all the more complicated stuff to a different (non-default) keymap would be fine.

File: keyboards/handwired/2d4d/info.json [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+    "maintainer": "qmk",
+    "manufacturer": "mechMehmet",
+    "development_board": "blackpill_f401",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x0000",
+        "vid": "0xFEED"
+    },
+    "features": {
+        "encoder": true,
```
Comment (waffle87) [inline keyboards/handwired/2d4d/info.json:12]: If encoders are enabled at the keyboard level, there needs to be some keyboard level encoder functionality as well. See [doc](https://docs.qmk.fm/#/feature_encoders?id=callbacks).

File: keyboards/handwired/2d4d/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,450 @@
...
+            writePinHigh(ledRed);
+            writePinHigh(ledBlue);
+            writePinHigh(ledGreen);
+            break;
+    }
+    return state;
+}
+ 
+
+bool led_update_kb(led_t led_state) {
```
Comment (waffle87) [inline keyboards/handwired/2d4d/keymaps/default/keymap.c:259]:
```suggestion
bool led_update_user(led_t led_state) {
```

**Pull request closed without merging**